code/game/objects/items/devices/geiger.dm

```diff
@@ -0,0 +1,41 @@
+#define COOLDOWN 1 SECONDS
+
+/obj/item/device/geiger_counter
+	name = "geiger counter"
+	desc = "a device about the size of a briefcase, used for detecting and measuring ambient radiation."
+	icon_state = "geiger_counter"
+	w_class = W_CLASS_LARGE
+	origin_tech = Tc_ENGINEERING + "=3;" + Tc_MATERIALS + "=4"
+	var/on = 0
+	var/last_call = 0
+	var/event_key
+
+/obj/item/device/geiger_counter/New()
+	..()
+	update_icon()
+
+/obj/item/device/geiger_counter/pickup(mob/user)
+	event_key = user.on_irradiate.Add(src, "measure_rad")
+
+/obj/item/device/geiger_counter/dropped(mob/user)
+	user.on_irradiate.Remove(event_key)
+	event_key = null
+
+
+/obj/item/device/geiger_counter/proc/measure_rad(list/arguments)
+	var/mob/user = arguments["user"]
+	var/rads = arguments["rads"]
+	if(on && world.time > last_call + COOLDOWN)
+		to_chat(user, "<span class = 'notice'>Radiation detected.</span>")
+		spawn(5)
```

Comment:
This will still proc more than the COOLDOWN would allow due to the last_call being set after the spawn(5)

Set it before the spawn(5) or add the `&& world.time > last_call + COOLDOWN` below